lib/ts/nextjs/ssr.ts

```diff
@@ -0,0 +1,218 @@
+
+const CURRENT_PATH_COOKIE_NAME = "sCurrentPath";
+const FORCE_LOGOUT_PATH_PARAM_NAME = "forceLogout";
+const REDIRECT_PATH_PARAM_NAME = "stRedirectTo";
+
+type SSRSessionState =
+    | "front-token-not-found"
+    | "front-token-invalid"
+    | "front-token-expired"
+    | "access-token-not-found"
+    | "access-token-invalid"
+    | "tokens-do-not-match"
+    | "tokens-match";
+
+export default class SuperTokensNextjsSSRAPIWrapper {
+    static config: SuperTokensNextjsConfig;
+
+    static init(config: SuperTokensNextjsConfig) {
+        if (config.enableDebugLogs) {
+            enableLogging();
+        }
+        SuperTokensNextjsSSRAPIWrapper.config = config;
+    }
+
+    static getConfigOrThrow(): SuperTokensNextjsConfig {
+        if (!SuperTokensNextjsSSRAPIWrapper.config) {
+            throw new Error("SuperTokens must be initialized before calling this method.");
+        }
+
+        return SuperTokensNextjsSSRAPIWrapper.config;
+    }
+
+    /**
+     * Get the session state or refresh the authentication for a Next.js App Router project
+     * @param cookies - The cookies store exposed by next/headers (await cookies())
+     * @param redirect - The redirect function exposed by next/navigation
+     * @returns The session context value or directly redirects the user to either the login page or the refresh API
+     **/
+    static async getSSRSession(cookies: CookiesStore, redirect: (url: string) => never): Promise<LoadedSessionContext> {
+        const redirectPath = cookies.get(CURRENT_PATH_COOKIE_NAME)?.value || "/";
+        const authPagePath = `${getAuthPagePath()}?${REDIRECT_PATH_PARAM_NAME}=${redirectPath}`;
+        const refreshLocation = `${getRefreshLocation()}?${REDIRECT_PATH_PARAM_NAME}=${redirectPath}`;
+
+        const { state, session } = await getSSRSessionState(cookies);
+        logDebugMessage(`SSR Session State: ${state}`);
+        switch (state) {
+            case "front-token-not-found":
+            case "front-token-invalid":
+            case "access-token-invalid":
+                logDebugMessage(`Redirecting to Auth Page: ${authPagePath}`);
+                return redirect(authPagePath);
+            case "front-token-expired":
+            case "access-token-not-found":
+            case "tokens-do-not-match":
+                logDebugMessage(`Redirecting to refresh API: ${refreshLocation}`);
+                return redirect(refreshLocation);
+            case "tokens-match":
+                logDebugMessage("Returning session object");
+                return session as LoadedSessionContext;
+            default:
+                // This is here just to prevent typescript from complaining
+                // about the function not returning a value
+                throw new Error(`Unknown state: ${state}`);
+        }
+    }
+
+    // TODO: This method is isolated atm to make it easier to test and debug
+    // In the end it should be merged in the getSSRSession function
```

Comment:
Why do we need to merge it into getSSRSession? At first glance this works well as both the usecase and the impl is different enough.